cpp/ql/src/Security/CWE/CWE-078/ExecTainted.ql

```diff
@@ -15,63 +15,113 @@
 import cpp
 import semmle.code.cpp.security.CommandExecution
 import semmle.code.cpp.security.Security
-import DataFlow::PathGraph
 import semmle.code.cpp.ir.dataflow.TaintTracking
+import semmle.code.cpp.ir.dataflow.TaintTracking2
 import semmle.code.cpp.ir.IR
 import semmle.code.cpp.security.FlowSources
+import semmle.code.cpp.models.implementations.Strcat
 
 Expr sinkAsArgumentIndirection(DataFlow::Node sink) {
-  result = sink.asOperand()
-      .(SideEffectOperand)
-      .getAddressOperand()
-      .getAnyDef()
-      .getUnconvertedResultExpression()
+  result =
+    sink.asOperand()
+        .(SideEffectOperand)
+        .getAddressOperand()
+        .getAnyDef()
+        .getUnconvertedResultExpression()
 }
 
-class ExecTaintConfiguration extends TaintTracking::Configuration {
-  ExecTaintConfiguration() { this = "ExecTaintConfiguration" }
+predicate interestingConcatenation(DataFlow::Node fst, DataFlow::Node snd) {
```

Comment:
This predicate badly needs an explanation of what it's doing, especially the `not literal.getConvSpecOffset(index) = 0` line.  I can see from the context of this PR and the commit comment, but future maintainers won't (easily) be able to see that.

Reply:
Added